cache/async_cache_test.go

```diff
@@ -60,30 +62,32 @@ func TestAsyncCache_AwaitForConcurrentTransaction_GraceTimeWithoutTransactionCom
 		Query: []byte("SELECT async cache AwaitForConcurrentTransaction"),
 	}
 
-	_, err := asyncCache.Status(key)
-	if !errors.Is(err, ErrMissingTransaction) {
+	status, err := asyncCache.Status(key)
+	assert.NoError(t, err)
+	if !status.IsAbsent() {
 		t.Fatalf("unexpected behaviour: transaction isnt done while it wasnt even started")
 	}
 
 	if err := asyncCache.Create(key); err != nil {
 		t.Fatalf("unexpected error: %s failed to register transaction", err)
 	}
 
-	status, err := asyncCache.Status(key)
-	if err != nil || !status.IsPending() {
+	status, err = asyncCache.Status(key)
+	assert.NoError(t, err)
+	if !status.IsPending() {
 		t.Fatalf("unexpected behaviour: transaction isnt finished")
 	}
 
 	startTime := time.Now()
-	transactionResult, err := asyncCache.AwaitForConcurrentTransaction(key)
-	if err != nil {
-		t.Fatalf("unexpected behaviour while awaiting concurrent transaction: %v", err)
-	}
+	_, err = asyncCache.AwaitForConcurrentTransaction(key)
+	assert.NoError(t, err)
 	elapsedTime := time.Since(startTime)
 
 	// in order to let the cleaner swipe the transaction
-	time.Sleep(100 * time.Millisecond)
-	if !transactionResult.State.IsPending() {
+	time.Sleep(150 * time.Millisecond)
```

Comment:
good catch!
[it's not related to this PR but it's a bit of a shame that the 100 msec frequency of the cleaner is hardcoded. Because it makes the 150 msec choice in this test a bit magical for a newcomer]

Reply:
I'll add a todo to refactor at some point.